### PR TITLE
resource_control: load resource unit config from local pd client

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: ref #38825

Problem Summary:

### What is changed and how it works?
- Fetch the `request-units` configs from local resource-control client instead of executing a `show config` query to avoid the requiring of `SYSTEM_VARIABLES_ADMIN` privilege.
- Add a check of `tidb_enable_resource_control` before execute `calibrate resource`

NOTE:
Currently, this PR put the pd-client `ResourceGroupController` under the executor package, I think this is not a very good idea but I can't find a better place. Maybe we should put all resource-group related logic into a separate package.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/calibrate_resource.go [hunk 1]
```diff
@@ -16,54 +16,70 @@ package executor
...
+)
+
+// SetResourceGroupController set a inited ResourceGroupsController for calibrate usage.
+func SetResourceGroupController(rc *rmclient.ResourceGroupsController) {
+	resourceGrooupCtl = rc
+}
+
+// GetResourceGroupController returns the ResourceGroupsController.
+func GetResourceGroupController() *rmclient.ResourceGroupsController {
+	return resourceGrooupCtl
```
Comment (HuSharp) [inline executor/calibrate_resource.go:82]: extra `o`

**Review by HuSharp — APPROVED** [review]
LGTM

**Review by nolouch — APPROVED** [review]
LGTM. BTW, do you ensure the `METRICS_SCHEMA` can be accessed?

**Comment by glorv** [comment]
> LGTM. BTW, do you ensure the `METRICS_SCHEMA` can be accessed?

@nolouch I have added privileges requirements in the docs. See: https://github.com/pingcap/docs-cn/pull/13714

**Comment by nolouch** [comment]
PTAL @JmPotato

**Review by JmPotato — APPROVED** [review]

**Pull request merged**